src/compas/datastructures/mesh/core/attributes.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+try:
+    from collections.abc import MutableMapping  # python > 3.3
+except ImportError:
+    from collections import MutableMapping      # python 2.7
+
+
+__all__ = ['VertexAttributeView', 'FaceAttributeView', 'EdgeAttributeView']
+
+
+class AttributeView(object):
+    """Mixin for attribute dict views."""
+
+    def __init__(self, defaults, attr, key, custom_only):
+        self.defaults = defaults
+        self.attr = attr
+        self.key = key
+        self.custom_only = custom_only
+        self.attr.setdefault(self.key, {})
```

Comment:
this is problematic for vertices. if a user checks for the data of a non-existing vertex it actually gets created... looking at the old implementation, there should be a guard for this as well, in order to not create non-existing edges or faces in the data dicts.

Reply:
Can we just suppress that line and let it raise a `KeyError` if `self.key` is not in `self.attr`? 

The other thing could be to omit `self.key` completely, and pass for directly i.e. `FaceAttributeView(attr=self.facedata[key])` as opposed to `FaceAttributeView(attr=self.facedata, key=key)`...

Reply:
yes but the line is needed for the edge data and face data views. there is not necessarily an entry in those dicts for existing edges and faces.

Reply:
in that sense `XAttributeView(attr=facedata[key])` will not work. unless we do something like `XAttributeView(attr=facedata.setdefault(key, {}))` or if we make sure that entries in the attr dicts are always added when the items are added, but that is not so easy to enforce and it creates unnecessary overhead in many algorithms...

Reply:
just to clarify, for me this init function cannot be used for vertices, because checking the attributes of a non-existing vertex would actually add it, and that would be an extremely unexpected side-effect.

Reply:
Makes sense. I will change that.